### PR TITLE
Only install netifaces for Humble and Iron.

### PR DESCRIPTION
**Pull request by clalancette**
In Rolling, we've switched to not needing netifaces anymore.

A draft until I run CI on it.

### CODE REVIEW

**Comment by clalancette** [comment]
CI for this is in https://github.com/ros2/ci/pull/748#issuecomment-1923903388

**Review by mjcarroll — APPROVED** [review]

File: cookbooks/ros2_windows/recipes/pip_installs.rb [hunk 1]
```diff
@@ -22,16 +21,21 @@
 ]
 
 ros2doctor_network_dependency = {
-  "foxy" => "ifcfg",
-  "galactic" => "ifcfg",
```
Comment (nuclearsandwich) [inline cookbooks/ros2_windows/recipes/pip_installs.rb:26]: Slightly unthrilled by the cleanup being lumped in with the explicit changes but not enough to block it.

File: cookbooks/ros2_windows/recipes/pip_installs.rb [hunk 2]
```diff
@@ -22,16 +21,21 @@
...
   "humble" => "psutil",
   "iron" => "psutil",
   "rolling" => "psutil",
 }.freeze
 
-
```
Comment (nuclearsandwich) [inline cookbooks/ros2_windows/recipes/pip_installs.rb:32]: Ditto the whitespace changes.

**Review by nuclearsandwich — APPROVED** [review]

**Comment by clalancette** [comment]
Oh, CI for this was in https://github.com/ros2/ci/pull/748#issuecomment-1923903388 and https://github.com/ros2/ci/pull/748#issuecomment-1936529660

**Pull request merged**